Assets/Scripts/PlayerController/PlayerAnimation.cs

```diff
@@ -49,4 +54,42 @@ void Update()
                 m_Animator.speed = 1;
         }
     }
+
+    public bool hasJumpPowerUp() 
+    {
+        return m_hasJumpPower;
+    }
+
+    public bool hasSpeedPowerUp()
+    {
+        return m_hasSpeedPower;
+    }
+
+    public void GoFaster()
+    {
+        StartCoroutine("DoubleUpSpeed");
+    }
+
+    public void JumpHigher()
+    {
+        StartCoroutine("DoubleUpJump");
+    }
+
+    IEnumerator DoubleUpJump()
+    {
+        m_hasJumpPower = true;
+        m_PlayerMovement.SetDoubleJump(true);
+        yield return new WaitForSeconds(m_doubleBuffDuration);
+        m_hasJumpPower = false;
+        m_PlayerMovement.SetDoubleJump(false);
+    }
+
+    IEnumerator DoubleUpSpeed()
+    {
+        m_hasSpeedPower = true;
+        m_PlayerMovement.SetDoubleSpeed(true);
+        yield return new WaitForSeconds(m_doubleBuffDuration);
+        m_hasSpeedPower = false;
+        m_PlayerMovement.SetDoubleSpeed(false);
+    }
```

Comment:
Good use of Coroutines. Maybe spend some time teaching this trick to the rest ;)